test/snippet/read_mapper/mapper_final.cpp

```diff
@@ -0,0 +1,150 @@
+        cereal::BinaryInputArchive iarchive{is};
+        iarchive(index);
+    }
+
+    seqan3::sequence_file_input<dna4_traits> query_file_in{arguments.query_path};
+
+    using sam_fields_t = seqan3::fields<seqan3::field::seq,
+                                        seqan3::field::id,
+                                        seqan3::field::ref_id,
+                                        seqan3::field::ref_offset,
+                                        seqan3::field::alignment,
+                                        seqan3::field::qual,
+                                        seqan3::field::mapq>;
+
+    seqan3::sam_file_output sam_out{arguments.sam_path, sam_fields_t{}};
+
+    namespace sc = seqan3::search_cfg;
+    seqan3::configuration const search_config = sc::max_error_total{sc::error_count{arguments.errors}}
+                                              | sc::hit_all_best{};
+
+    namespace ac = seqan3::align_cfg;
+    seqan3::configuration const align_config = ac::method_global{ac::free_end_gaps_sequence1_leading{true},
+                                                                 ac::free_end_gaps_sequence2_leading{false},
+                                                                 ac::free_end_gaps_sequence1_trailing{true},
+                                                                 ac::free_end_gaps_sequence2_trailing{false}}
+                                             | ac::edit_scheme
+                                             | ac::output_alignment{}
+                                             | ac::output_begin_position{}
+                                             | ac::output_score{};
+
+    for (auto && record : query_file_in)
+    {
+        auto const & query = record.sequence();
+        for (auto && result : search(query, index, search_config))
+        {
+            size_t const start = result.reference_begin_position() ? result.reference_begin_position() - 1 : 0;
+            std::span text_view{std::data(storage.sequences[result.reference_id()]) + start, query.size() + 1};
+
+            for (auto && alignment : seqan3::align_pairwise(std::tie(text_view, query), align_config))
+            {
+                auto const aligned_seq = alignment.alignment();
+                size_t const ref_offset = alignment.sequence1_begin_position() + 2 + start;
+                size_t const map_qual = 60u + alignment.score();
+
+                sam_out.emplace_back(query,
+                                     record.id(),
+                                     storage.ids[result.reference_id()],
+                                     ref_offset,
+                                     aligned_seq,
+                                     record.base_qualities(),
+                                     map_qual);
+            }
+        }
+    }
+}
+
+void run_program(cmd_arguments const & arguments)
+{
+    reference_store_t storage{};
+    read_reference(arguments.reference_path, storage);
```

Comment:
TODO: Wouldn't it be smarter to store the sequence IDs in the index (i.e. as part of the indexer executable), instead of reading references here again? Usually, you do not need to provide the reference to the mapper itself.

Reply:
No strong feelings, if it works without it, why not :)

Reply:
So you mean that we somehow store the reference sequences, that we need for the alignment, in the index itself?

Reply:
Yes,

we could just serialise the id-vector in the indexer and read it in the mapper.
We could either just serialise fm_index+id-vector explicitly into one file, or do a struct that encapsulates both. Former should be the easier one.